armory/paths.py

```diff
@@ -9,58 +9,132 @@
+            "Error parsing config.json. Please run `armory configure`.\n"
+            "    If you previously ran an older version of armory, you may\n"
+            f"    need to remove the {default().armory_dir} directory due to changes"
         )
+        raise
+
     return config
 
 
 class HostPaths:
+    def __init__(self):
+        self.cwd = default().cwd
+        self.user_dir = default().user_dir
+        self.armory_dir = default().armory_dir
+        self.armory_config = default().armory_config
+        if os.path.isfile(self.armory_config):
+            # Parse paths from config
+            config = load_config()
+            for k in "dataset_dir", "model_dir", "output_dir", "tmp_dir":
+                setattr(self, k, config[k])
+            self.external_repo_dir = os.path.join(self.dataset_dir, "external_repos")
+        else:
+            logger.warning(f"No {self.armory_config} file. Using default paths.")
+            logger.warning("Please run `armory configure`")
+            self.dataset_dir = default().dataset_dir
+            self.model_dir = default().model_dir
+            self.tmp_dir = default().tmp_dir
+            self.output_dir = default().output_dir
+            self.external_repo_dir = default().external_repo_dir
+
+    def makedirs(self):
+        logger.info("Creating armory directories if they do not exist")
+        os.makedirs(self.dataset_dir, exist_ok=True)
+        os.makedirs(self.model_dir, exist_ok=True)
+        os.makedirs(self.tmp_dir, exist_ok=True)
+        os.makedirs(self.output_dir, exist_ok=True)
+
+
+class HostDefault:
     def __init__(self):
         self.cwd = os.getcwd()
         self.user_dir = os.path.expanduser("~")
         self.armory_dir = os.path.join(self.user_dir, ".armory")
-        self.tmp_dir = os.path.join(self.armory_dir, "tmp")
         self.armory_config = os.path.join(self.armory_dir, "config.json")
 
-        # Parse paths from config
-        config = _parse_global_config(self.armory_config, self.armory_dir)
-        self.dataset_dir = config.get("cached_dataset_dir")
-        self.output_dir = config.get("output_dir")
+        self.dataset_dir = os.path.join(self.armory_dir, "datasets")
+        self.model_dir = os.path.join(self.armory_dir, "models")
+        self.tmp_dir = os.path.join(self.armory_dir, "tmp")
+        self.output_dir = os.path.join(self.armory_dir, "outputs")
         self.external_repo_dir = os.path.join(self.dataset_dir, "external_repos")
 
 
 class DockerPaths:
     def __init__(self):
-        self.armory_dir = "/armory"
-        self.tmp_dir = "/armory/tmp"
-        self.dataset_dir = "/datasets"
-        self.output_dir = "/outputs"
+        self.cwd = "/workspace"
+        armory_dir = "/armory"
+        self.dataset_dir = armory_dir + "/datasets"
+        self.model_dir = armory_dir + "/models"
+        self.tmp_dir = armory_dir + "/tmp"
+        self.output_dir = armory_dir + "/outputs"
+        # TODO: external_repo_dir?
```

Comment:
Currently external_repo just goes in dataset which is probably not the final place. Lets go ahead and add this as a path now so that later we don't need to parse configs and add it if it doesn't exist as a special case.

Reply:
Makes sense.

Since we download it each time we run, I think it makes sense to have this be in the `tmp` directory. Thoughts?

Reply:
Agree that works nicely